### ISSUE TITLE
Interface methods should not be required to have return statements

### ISSUE THREAD

**Issue by tschaub** [eslint/eslint#4887]
With ESLint @65ad608ca3bb71a45a3b14a6f8dc632ea86d2949, the `valid-jsdoc` rule with `requireReturn: false` fails if a function has a `@return` tag but no `return` statement.

For a constructor/class with `@interface`, methods may be empty functions (and this is may be a requirement).  So the following is not valid according to `valid-jsdoc`:
```js
/* eslint valid-jsdoc: [2, {requireReturn: false}] */

/**
 * Geometry interface.
 * @interface
 */
function Geometry() {}

/**
 * Get the geometry type.
 * @return {string} The geometry type.
 */
Geometry.prototype.getType = function() {};
```
>   10:1  error  Unexpected @returns tag; function has no return statement  valid-jsdoc

I'm curious if others think it would be worth updating `valid-jsdoc` to handle this case.  It feels like there may be a fair bit of work involved to create a lookup of all `@interface` and then ignore the `return` statement check if there is a member expression with a prototype in the assignment statement.

The other alternative is to disable `valid-jsdoc` for this case.  Let me know if anybody else has other suggestions.

**Comment by eslintbot** [comment]
@tschaub Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by nzakas** [comment]
This was already implemented: https://github.com/eslint/eslint/pull/4861

Please be sure to search old issues prior to opening new ones.

**Comment by tschaub** [comment]
@nzakas are you sure you meant #4861?

That issue is about making it so functions marked with `@interface` aren't required to have a `return` statement.  The difference is subtle but significant.  This is about interface *methods*.  These are not marked with `@interface`.  They are prototype methods of a constructor that is marked with `@interface`.

Please see the example above for additional detail.  The issue is that the `Geometry.prototype.getType` method has no body - this is a requirement of the Closure Compiler because `Geometry` is an `@interface`.  So `valid-jsdoc` complains that it doesn't have a `return` statement.

**Comment by nzakas** [comment]
Ah, my apologies, I did misread your description. 

Unfortunately, there isn't a way to know that the method is part of an interface vs. a regular function. We might be able to have an option that allows empty functions, though.

**Comment by nzakas** [comment]
Unfortunately, it looks like consensus couldn't be reached on this issue and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after 21 days tend never to reach consensus, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.
